Views/backend/connect/controller/main.js

```diff
@@ -174,10 +179,96 @@ Ext.define('Shopware.apps.Connect.controller.Main', {
                 });
                 break;
             default:
-                me.mainWindow = me.getView('main.Window').create({
-                    'action': me.subApplication.action
-                }).show();
+                var recreateConnectCategories;
+                me.sendAjaxRequest(
+                    '{url controller=Connect action=checkForDataMigration}',
+                    {},
+                    function(response) {
+                        recreateConnectCategories = response.data.recreateConnectCategories;
+                        if (recreateConnectCategories) {
+                            Ext.Msg.show({
+                                    title: me.messages.applyMigrationsTitle,
+                                    msg: me.messages.applyMigrationsMessage,
+                                    buttons: Ext.Msg.YESNO,
+                                    icon: Ext.Msg.QUESTION,
+                                    fn: me.applyMigrations,
+                                    scope: me
+                                }
+                            );
+                        } else {
+                            me.mainWindow = me.getView('main.Window').create({
+                                'action': me.subApplication.action
+                            }).show();
+                        }
+                    }
+                );
                 break;
+
+        }
+    },
+
+    startMigration: function (window, offset) {
+        offset = parseInt(offset) || 0;
+        var batchsize = 100;
+        var limit = offset + batchsize;
```

Comment:
I guess you meant ```var articlesProccessed``` ? limit must be constant value, you already use batchSize for that